common/types/types.go

```diff
@@ -87,14 +87,19 @@ type LocalUser struct {
 //                    account to communicate with LDAP/AD. Hence this account
 //                    must have appropriate privileges, specifically for lookup.
 //  ServiceAccountPassword: of the service account
+//  StartTLS: if set, the connection will be upgrated to SSL/TLS mode
+//  TLSCertIssuedTo: Servername for which the TLS/SSL certificate was issued.
+//                   if this is empty, the certs will not be verified and
+//                   the connection is prone to man-in-the-middle attacks.
+//                   This is used only when `StartTLS` is enabled.
 type LdapConfiguration struct {
 	Server                 string `json:"server"`
 	Port                   uint16 `json:"port"`
 	BaseDN                 string `json:"base_dn"`
 	ServiceAccountDN       string `json:"service_account_dn"`
 	ServiceAccountPassword string `json:"service_account_password,omitempty"`
 	StartTLS               bool   `json:"start_tls"`
-	InsecureSkipVerify     bool   `json:"insecure_skip_verify"`
+	TLSCertIssuedTo        string `json:"tls_cert_issued_to"`
```

Comment:
I might call this `ExpectedCommonName` or something since that's really what it is.  It holds the Common Name you expect to be presented in the cert when you connect